sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -334,26 +331,23 @@ def _validate(self):
     @classmethod
     def _from_generated(cls, key_value):
         # type: (KeyValue) -> SecretReferenceConfigurationSetting
-        try:
-            if key_value is None:
-                return None
-            if key_value.value:
-                try:
-                    key_value.value = json.loads(key_value.value)
-                except json.decoder.JSONDecodeError:
-                    pass
-            return cls(
-                key=key_value.key,
-                secret_uri=key_value.value[u"secret_uri"],
-                label=key_value.label,
-                secret_id=key_value.value,
-                last_modified=key_value.last_modified,
-                tags=key_value.tags,
-                read_only=key_value.locked,
-                etag=key_value.etag,
-            )
-        except (KeyError, AttributeError):
-            return ConfigurationSetting._from_generated(key_value)
+        if key_value is None:
+            return None
+        if key_value.value:
+            try:
+                key_value.value = json.loads(key_value.value)
+            except json.decoder.JSONDecodeError:
+                pass
```

Comment:
Why pass not raise ValueError?

Reply:
When we try to do the indexing it will raise a `TypeError` that is caught, and a `ConfigurationSetting` is returned instead.

Reply:
My concern is if we pass here, will it get mad in key=key_value.key?

Reply:
This is called by the `ConfigurationSetting._from_generated` method so if it does raise an error on `key=key_value.key`, the ConfigurationSetting will catch the error and return a plain `ConfigurationSetting` object.